pkg/build/build.go

```diff
@@ -99,14 +99,15 @@ func (c *Build) BuildImage(dockerfile, image, context string) error {
 		Dockerfile:   dockerfile,
 	}
 
+	// Delete tarball after creating image
+	defer os.Remove(tmpFile.Name())
```

Comment:
defer should be placed right after `tmpFile` is created. Otherwise, it won't be run before for any return between this and  	`tmpFile, err := ioutil.TempFile("/tmp", "kedge-image-build-")`